types/ts3.9.4/index.d.ts

```diff
@@ -23,6 +24,11 @@ import { DefaultParams, Params, Match, MatcherFn } from "../matcher";
 export * from "../matcher";
 export * from "../use-location";
 
+// React <18 only: fixes incorrect `ReactNode` declaration that had `{}` in the union.
```

Comment:
Side thought: honestly, supporting two versions of TS becomes a bit of a burden. I wonder if we could drop support for 3.9.4 in the future major version. Would that be too brutal? Imo, it could, but there is an alternative – can't upgrade to TS 4, well you would have to stay on wouter@2, sorry.

Reply:
🤷‍♂️  I have no idea. Personally, I don't use TS < 4 and never plan to. But others may have a different opinion on this 😄